tests/network/openvpn_server.pm

```diff
@@ -0,0 +1,128 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test OpenVPN on two machines - this one is server.
+# Maintainer: Pavel Dostál <pdostal@suse.cz>
+
+use base 'consoletest';
+use testapi;
+use lockapi;
+use mm_network;
+use mmapi 'wait_for_children';
+use utils qw(systemctl zypper_call exec_and_insert_password);
+use strict;
+
+sub run {
+    my ($self) = @_;
+    barrier_create('OPENVPN_STATIC_START',    2);
+    barrier_create('OPENVPN_STATIC_STARTED',  2);
+    barrier_create('OPENVPN_STATIC_FINISHED', 2);
+    barrier_create('OPENVPN_CA_START',        2);
+    barrier_create('OPENVPN_CA_STARTED',      2);
+    barrier_create('OPENVPN_CA_FINISHED',     2);
+    select_console "root-console";
+
+    # Configure static network, disable firewall
+    configure_default_gateway;
+    systemctl 'stop ' . $self->firewall;
+    configure_static_ip('10.0.2.101/24');
+    configure_static_dns(get_host_resolv_conf());
+
+    # Install openvpn, generate static key
+    zypper_call('in openvpn');
+    assert_script_run('cd /etc/openvpn');
+    assert_script_run('openvpn --genkey --secret static.key');
+    mutex_create 'OPENVPN_STATIC_KEY';
+
+    # Write the server config
+    assert_script_run(qq(echo "dev tun
+ifconfig 10.8.0.1 10.8.0.2
+secret /etc/openvpn/static.key" > static.conf));
+
+    # Start the server
+    systemctl('start openvpn@static');
+    systemctl('status openvpn@static -l');
+    barrier_wait 'OPENVPN_STATIC_START';
+
+    # Test the connection when also the client is ready
+    barrier_wait 'OPENVPN_STATIC_STARTED';
+    assert_script_run("ping -c5 -W1 -I tun0 10.8.0.2");
+
+    # Stop the server when also client is done
+    barrier_wait 'OPENVPN_STATIC_FINISHED';
+    systemctl('stop openvpn@static');
+
+    assert_script_run("curl -Lo easy-rsa.tgz https://github.com/OpenVPN/easy-rsa/releases/download/v3.0.5/EasyRSA-nix-3.0.5.tgz");
+    assert_script_run("tar xvf easy-rsa.tgz");
+    assert_script_run("mv EasyRSA-3.0.5 easy-rsa");
+
+    assert_script_run("./easy-rsa/easyrsa init-pki");
+    assert_script_run("ls pki/");
+
+    assert_script_run("./easy-rsa/easyrsa gen-dh", 180);
+    assert_script_run("ls pki/dh.pem");
+
+    type_string "./easy-rsa/easyrsa build-ca nopass\n";
+    sleep 3;
+    send_key 'ret';
+    assert_script_run("ls pki/ca.crt");
+
+    type_string "./easy-rsa/easyrsa gen-req server nopass\n";
+    sleep 3;
+    send_key 'ret';
```

Comment:
yes can use `assert_script_run "yes | ./easy-rsa/easyrsa build-ca nopass"`

Reply:
Yes, it's done now :+1: [server](http://pdostal-server.suse.cz/tests/212) [client](http://pdostal-server.suse.cz/tests/213)